acceptance/tests/catalog_with_binary_data.rb

```diff
@@ -0,0 +1,72 @@
+require 'puppet/acceptance/common_utils'
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+test_name "C100300: Catalog containing binary data is applied correctly" do
+  skip_test 'requires a master for serving module content' if master.nil?
+  tag 'risk:medium'
+
+  test_num = 'c100300'
+  tmp_environment = mk_tmp_environment_with_teardown(master, File.basename(__FILE__, '.*'))
+  agent_tmp_dirs = {}
+  agents.each do |agent|
+    agent_tmp_dirs[agent.hostname] = agent.tmpdir(tmp_environment)
+  end
+
+  teardown do
+    step 'remove all test files on agents' do
+      agents.each { |agent| on(agent, "rm -r #{agent_tmp_dirs[agent.hostname]}", :accept_all_exit_codes => true) }
+    end
+    # note - master teardown is registered by #mk_tmp_environment_with_teardown
+  end
+
+  step "Create module with binary data file on master" do
+    on(master, "mkdir -p #{environmentpath}/#{tmp_environment}/modules/#{test_num}/{manifests,files}")
+    master_module_manifest = "#{environmentpath}/#{tmp_environment}/modules/#{test_num}/manifests/init.pp"
+    master_module_binary_file = "#{environmentpath}/#{tmp_environment}/modules/#{test_num}/files/binary_data"
+
+    bin_content = '\xC0\xFF'
```

Comment:
hrm.
When I use double quotes, `binwrite` reports that the encoding is invalid.
```
    ArgumentError: invalid byte sequence in UTF-8
```

Using single quotes, the `binwrite`, and the test succeeds as expected.

Tested on RHEL-7 and Windows 2012